### ISSUE TITLE
Fix Prop reads and reps method

### ISSUE THREAD

**Issue by gcuster1991** [MayaGans/CoreMicro#14]
Currently the prop reads and reps method only accounts for reps. We need to edit the function to account for number of reads as well.

**Comment by MayaGans** [comment]
Currently: 

```
prop_reads_and_reps <- function(otu_table, sites = 0.5, taxa_as_rows = TRUE) {

  # transpose data if rows are not taxa
  if (!taxa_as_rows) otu_table <- transpose_taxa(otu_table)

  # rename first column `X`
  names(otu_table)[1] <- "X"

  otu_table %>%
    tidyr::pivot_longer(-X) %>%
    dplyr::group_by(X) %>%
    dplyr::mutate(row_sum = sum(value)) %>%
    dplyr::ungroup() %>%
    dplyr::mutate(s = sum(unique(row_sum))) %>%
    dplyr::filter(value > 0) %>%
    dplyr::group_by(X) %>%
    dplyr::summarise(
      num_sites = n(),
      N = ncol(otu_table) - 1) %>%
    dplyr::filter(num_sites >= N * sites) %>%
    dplyr::pull(X)
}
```

I want to add a `total_reads = 0.01` param but I'm not sure where this would go.... I think here `    dplyr::filter(value > 0)` but still not *quite* sure what it would look like.... 

Any thoughts @gcuster1991 ?

**Comment by MayaGans** [comment]
```
prop_reads_and_reps <- function(otu_table, sites = 0.5, total_reads = 0.01, taxa_as_rows = TRUE) {

  # transpose data if rows are not taxa
  if (!taxa_as_rows) otu_table <- transpose_taxa(otu_table)

  # rename first column `X`
  names(otu_table)[1] <- "X"

  otu_table %>%
    tidyr::pivot_longer(-X) %>%
    dplyr::group_by(X) %>%
    dplyr::mutate(row_sum = sum(value)) %>%
    dplyr::ungroup() %>%
    dplyr::mutate(s = sum(unique(row_sum))) %>%
    # this seems wrong.....
    dplyr::filter(value > total_reads*s) %>% 
    dplyr::group_by(X) %>%
    dplyr::summarise(
      num_sites = n(),
      N = ncol(otu_table) - 1) %>%
    dplyr::filter(num_sites >= N * sites) %>%
    dplyr::pull(X)
}
```

**Comment by gcuster1991** [comment]
Let me check how we did it in the old notebook. I think it was something like num_reads_to_taxon/total_reads > 0.02. 

should it be included here:
dplyr::filter(num_sites >= N * sites & num_reads>0.02*total_reads)

**Comment by MayaGans** [comment]
This.... Does not seem right:

```{r}
  otu_table %>%
    tidyr::pivot_longer(-X) %>%
    dplyr::group_by(X) %>%
    dplyr::mutate(num_reads = sum(value)) %>%
    dplyr::ungroup() %>%
    dplyr::group_by(X, num_reads) %>%
    dplyr::mutate(s = sum(num_reads)) %>%
    dplyr::ungroup() %>%
    dplyr::filter(value > 0) %>%
    dplyr::group_by(X) %>%
    dplyr::mutate(
      # what is this supposed to be?
      num_sites = n(),
      # what about this?
      N = ncol(otu_table) - 1) %>%
    dplyr::ungroup() %>%
    dplyr::filter(num_sites >= N * propreps & num_reads > propreads*s) %>%
    dplyr::pull(X)
```

```
X         name                   value num_reads       s num_sites     N
   <chr>     <chr>                  <dbl>     <dbl>   <dbl>     <int> <dbl>
 1 OTU_14834 M21.Col.1R.yng.M1_27b      9      9312 2681856       287   288
 2 OTU_14834 M21.Col.2R.yng.M1_27b     26      9312 2681856       287   288
 3 OTU_14834 M21.Col.3R.yng.M1_27b      2      9312 2681856       287   288
 4 OTU_14834 M21.Col.4R.yng.M1_27b      4      9312 2681856       287   288
 5 OTU_14834 M21.Col.5R.yng.M1_27b      8      9312 2681856       287   288
 6 OTU_14834 M21.Col.6R.yng.M1_27b     34      9312 2681856       287   288
 7 OTU_14834 M21.Col.8R.yng.M1_27b      8      9312 2681856       287   288
 8 OTU_14834 M21.Col.10R.yng.M1_27b    23      9312 2681856       287   288
 9 OTU_14834 M21.Ct.1R.yng.M1_27b      12      9312 2681856       287   288
10 OTU_14834 M21.Ct.2R.yng.M1_27b      12      9312 2681856       287   288
```

N and num_sites are only 1 off from one another?

**Comment by MayaGans** [comment]
Okay I am certain this is right until the last two filters (where we actually apply the method..... Thoughts?

```
otu_table %>%
    tidyr::pivot_longer(-X) %>%
    dplyr::group_by(X) %>%
    # count the number of reads of each OTU across sites
    dplyr::mutate(num_reads = sum(value)) %>%
    dplyr::add_count(X) %>%
    dplyr::ungroup() %>%
    # get total number of reads for every unique OTU
    dplyr::mutate(s = sum(num_reads)/n) %>%
    dplyr::ungroup() %>%
    # remove the rows with no values
    # so we can count the number of sites each OTU is present in
    dplyr::filter(value > 0) %>%
    dplyr::group_by(X) %>%
    dplyr::mutate(
      # sites per OTU
      num_sites = n(),
      # total sites
      N = ncol(otu_table) - 1) %>%
    dplyr::ungroup() %>%
    # this one gives us results
    dplyr::filter(num_sites >= N * propreps) %>%
    # this leaves us with nothing...
    dplyr::filter(num_reads >= propreads*s) %>%
    dplyr::pull(X)
```

**Comment by MayaGans** [comment]
What is `num_reads_to_taxon` here

**Comment by MayaGans** [comment]
Closes with PR #15
